linkedlist.c

```diff
@@ -0,0 +1,156 @@
+/* Jared Wasserman -- linkedlist.c */
+/* This program has various function for a linked list such as adding, removing, printing, comparing, and clearing the whole list.*/
+#include <stdio.h>
+#include <stdlib.h> 
+
+/* Node */
+typedef struct node{
+	int data;
+	struct node *next;
+}node;
+
+/* add */
+node* add(node *pointer, int data){
+	node *newPtr = (struct node*) malloc (sizeof(struct node));
+	newPtr->next = pointer;
+	newPtr->data = data;
+	return newPtr;
+}
+
+/* printList */
+void printList(node *ptr){
+	while(ptr->next!=NULL){
+		printf("%d\n",(*ptr).data);
+		ptr = ptr->next;
+	}
+}
+
+/* delete */
+void delete(node *pointer, int data){
+	while(pointer->next!=NULL && (pointer->next)->data != data){ 
+		pointer = pointer -> next;
+	}
+	if(pointer->next==NULL){
+	    printf("%d was not found\n",data);
+        return;
+	}
+	node *tempPtr = pointer -> next;
+	pointer->next = tempPtr->next;
+	free(tempPtr);
+	return;
+}
+
+/* clear */
+void clear(node *ptr){
+	node *temp;
+	while(ptr->next!=NULL){
+		temp=ptr;
+		free(ptr);
+		ptr=temp;
+		ptr=ptr->next;
+	}
+}
+
+/* smallestLength */
+int smallestLength(node *ptr, node *ptr2){
```

Comment:
for the `smallestLength` and `smallestSum` functions, you could have just done one while loop though! how would this algorithm look then?